packages/sanity/src/_internal/cli/actions/schema/storeSchemasAction.ts

```diff
@@ -0,0 +1,136 @@
+import {readFileSync} from 'node:fs'
+import path, {join, resolve} from 'node:path'
+
+import {type CliCommandArguments, type CliCommandContext} from '@sanity/cli'
+import chalk from 'chalk'
+
+import {
+  type CreateManifest,
+  type ManifestSchemaType,
+  type ManifestWorkspaceFile,
+} from '../../../manifest/manifestTypes'
+import {MANIFEST_FILENAME} from '../manifest/extractManifestAction'
+import {SANITY_WORKSPACE_SCHEMA_ID} from './schemaListAction'
+
+export interface StoreManifestSchemasFlags {
+  'path'?: string
+  'workspace'?: string
+  'id-prefix'?: string
+  'schema-required'?: boolean
+  'verbose'?: boolean
+}
+
+export default async function storeSchemasAction(
+  args: CliCommandArguments<StoreManifestSchemasFlags>,
+  context: CliCommandContext,
+): Promise<Error | undefined> {
+  const flags = args.extOptions
+  if (typeof flags.path === 'boolean') throw new Error('Path is empty')
+  if (typeof flags['id-prefix'] === 'boolean') throw new Error('Id prefix is empty')
+  if (typeof flags.workspace === 'boolean') throw new Error('Workspace is empty')
+
+  const schemaRequired = flags['schema-required']
+  const workspaceName = flags.workspace
+  const idPrefix = flags['id-prefix']
+  const verbose = flags.verbose
+  const {output, workDir, apiClient} = context
+
+  const defaultOutputDir = resolve(join(workDir, 'dist'))
+
+  const outputDir = resolve(defaultOutputDir)
+  const defaultStaticPath = join(outputDir, 'static')
+
+  const staticPath = flags.path ?? defaultStaticPath
+
+  const spinner = output.spinner({}).start('Storing schemas')
+
+  try {
+    const manifestPath = path.resolve(process.cwd(), staticPath)
+    const client = apiClient({
+      requireUser: true,
+      requireProject: true,
+    }).withConfig({apiVersion: 'v2024-08-01'})
+
+    const projectId = client.config().projectId
+
+    let manifest: CreateManifest
+
+    try {
+      manifest = JSON.parse(readFileSync(`${manifestPath}/${MANIFEST_FILENAME}`, 'utf-8'))
+    } catch (error) {
+      spinner.fail(`Manifest not found at ${manifestPath}/${MANIFEST_FILENAME}`)
+      output.error(error)
+      throw error
+    }
+
+    let storedCount = 0
+
+    let error: Error | undefined
+
+    const saveSchema = async (workspace: ManifestWorkspaceFile) => {
+      const id = `${idPrefix ? `${idPrefix}.` : ''}${SANITY_WORKSPACE_SCHEMA_ID}.${workspace.name}`
+      try {
+        if (workspace.projectId !== projectId) {
```

Comment:
I think we should leave a comment here saying something to the effect of "schema store does not support multi-project workspaces for now. We will add support for passing multi-project tokens in the future, and remove this check then. default cli tokens are project bound and will not be able to store for multi projects)